resources/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,452 @@
+| Use the custom app platform to build my application  | Learn, define and consume the APIs | Easy to use, adopt, low concept count | 
+| Manage and deploy the custom resources in the application platform| Create a IaC template for the custom resource type with compliance and other policies | Easy to author, test and deploy |  
+| Maintain the application over time | Versioning of the custom resources and enabling automatic upgrades | Less overhead to maintain |
+| Troubleshoot the custom application platform | Tools or experiences in Dashboard that enables users to troubleshoot and manage the custom resources | Immediate access to key data for troubleshooting, simple and intuitive |
+| Open sourcing templates | Repository and pipeline to validate, testing the infrastructure and publishing  | Desire to contribute, easy to use, low overhead /dependency to publish |
+
+## Solution approach in Radius
+
+### Main job : Build a custom platform for application development
+
+#### Define the set of key technologies for the enterprise applications
+Enterprises choose technologies based on the nature of technology, open-source support, documentation support, existing vendor relations, expertise in the team. Radius can provide the following to the users:
+
+*Community supported assets* : Community supported assets include the resource type definitions and IaC templates to deploy and manage resources in applications. Radius can enable the community to provide and use the assets for the prevalent cloud native technologies. This will help users to use pre-existing assets IaC templates and definitions in their applications without having to author them from scratch. 
+
+*Community supported assets meets their requirements* : Enterprises have a wide range of requirements for their applications based on the nature of the application, market, region, hosting models,compliance and security policies. Radius can enable the community to provide and use the resource type definitions and IaC templates that meet the requirements of the enterprises and meet them where they are. This will enable users to build what they want via Radius thus unblocking them to build their applications.
+
+*Maintenance and support for the community supported assets* : Radius can provide the necessary tools and services to maintain and support the community supported assets. This will help the community to contribute, maintain the assets and provide support to the users who are using the assets.
+
+**Solution in Radius**
+
+1. Resource extensibility for defining the custom resource types in Radius
+1. Recipes for the custom resource types
+1. Community supported resource type definitions
+1. Community supported Recipes
+1. Radius auto generates AWS, Azure Recipes from the definitions (Recipes for any resource) 
+
+We will cover the design and user experience for all the above solutions except for "Recipes for any resource" in this document to keep the scope of the work simple.
```

Comment:
First off, all of the content in this section is good. This is the right idea and we should do these things. 

----

However, I think all of this content goes with `Specify the contract for the key services in the application platform`. It's supporting work we do to help users specify the contract. My feedback is about the organization of the information and not about the ideas 👍 

In terms of solution - I don't think we can/should contribute much to `Define the set of key technologies for the enterprise applications`. Organizations have typically identified their "stack" independent of Radius, and those decisions are motivated by concerns separate from Radius. I think it's enough for us to mention that here.

Reply:
Updated the content to reflect the feedback